paddle/phi/infermeta/spmd_rules/rules.cc

```diff
@@ -753,6 +753,15 @@ PD_REGISTER_SPMD_RULE(nonzero,
 // add_n
 PD_REGISTER_SPMD_RULE(add_n, PD_INFER_SPMD(phi::distributed::AddNInferSpmd));
 
+// cummax
+PD_REGISTER_SPMD_RULE(cummax,
+                      PD_INFER_SPMD(phi::distributed::TopkInferSpmdBase),
+                      PD_INFER_SPMD(phi::distributed::TopkGradInferSpmdBase));
```

Comment:
Even if `CummaxInferSpmd` actually calls `TopkInferSpmdBase`, it is best to write `CummaxInferSpmd` instead of `TopkInferSpmdBase` when registering. and the parameters of `CummaxInferSpmd` and `TopkInferSpmdBase` are different. same issue with `CummaxGradInferSpmd` and `TopkGradInferSpmdBase`.